feat: add permission checks in dashboard and catalogue templates

mfarhan943 opened the pull request:
**Description**
Add permission checks in dashboard and catalogue templates

Related PR:
https://github.com/django-oscar/django-oscar/pull/4507

mfarhan943 committed:
feat: add permission checks in dashboard and catalogue templates

mfarhan943 commented on src/oscar/templates/oscar/dashboard/catalogue/product_row_actions.html:
```diff
@@ -16,5 +17,13 @@
                 {% trans "Delete" %}
             </a>
         </div>
+                {% has_dashboard_permission "catalogue-product" "catalogue_dashboard" as can_edit_product %}
```
Updated

mfarhan943 commented on src/oscar/templates/oscar/dashboard/catalogue/category_row_actions.html:
```diff
@@ -6,22 +6,35 @@
             {% trans "Actions" %}
         </button>
         <div class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuButton">
+            {% load dashboard_tags %}
```
Updated: [tables.py](https://github.com/django-oscar/django-oscar/pull/4534/files#diff-2d9f490a7c97ef99957083ec0a6f2640af655d19de17a62e29962d9c77e61936R55)

samar-hassan commented on src/oscar/apps/dashboard/catalogue/tables.py:
```diff
@@ -89,6 +108,21 @@ class CategoryTable(DashboardTable):
     icon = "sitemap"
     caption = ngettext_lazy("%s Category", "%s Categories")
 
+    def render_name(self, value, record):
```
Users with view category permission should be able to view all details of category, this will result in them being able to see only the fields that are displayed in the table. Maybe show the same category change view but disable the form fields and hide the change/update button.

samar-hassan replied:
I prefer 1, we can just hide/disable the update button, and also add a check in the post method

mfarhan943 committed:
refactor: move load tag to the top of the file

samar-hassan commented:
@mfarhan943 Sorry for the delay in getting back to this after a few months. Could you please take a look at the merge conflicts when you have a moment?

mfarhan943 commented:
> @mfarhan943 Sorry for the delay in getting back to this after a few months. Could you please take a look at the merge conflicts when you have a moment?

Done

samar-hassan commented on src/oscar/templates/oscar/dashboard/catalogue/product_row_actions.html:
```diff
@@ -1,20 +1,29 @@
+{% load dashboard_tags %}
 {% load i18n %}
+
+{% has_dashboard_permission "catalogue-product" "catalogue_dashboard" as can_change_product %}
```
Similar to how its done in this template, please move the `has_dasboard_permission` tags in all of the other templates as well i.e, in the beginning right after you load all the template tags.

samar-hassan commented on src/oscar/templates/oscar/dashboard/index.html:
```diff
@@ -158,7 +170,6 @@
         </table>
     </div>
     <div class="col-md-6">
-        {% if user.is_staff %}
```
Why did you remove this condition?

mfarhan943 committed:
fix: implement permission checks for dashboard offer/voucher stats

mfarhan943 committed:
feat: Add read-only permission checks for dashboard catalogue views

- Implement ReadOnlyUpdateViewMixin to enforce view-only restrictions on catalogue views.
- Fix permission bypass by checking is_superuser instead of is_staff.
- Hide action/save buttons and disable fields in templates when page is read-only.
- Prepend 'View' prefix to update titles in read-only mode.
- Control active offers & vouchers pane visibility on index index.

mfarhan943 committed:
fix: Allow partner dashboard users to bypass read-only view restrictions

- Exclude users with partner.dashboard_access permission from the read-only constraints in ReadOnlyUpdateViewMixin.get_readonly_status().
- Fixes the TestANonStaffUser::test_can_create_and_continue_editing_a_product functional test failure.
